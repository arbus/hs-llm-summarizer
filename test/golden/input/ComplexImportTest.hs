module ComplexImportTest where

-- Example with multiple imports, some with selectors
import Data.Text qualified as T
import Data.Map.Strict qualified as Map (empty, insert, lookup)
import Control.Monad.Reader qualified
import qualified Network.HTTP.Client as HTTP
import Data.Vector qualified
import Data.Aeson (ToJSON, FromJSON, (.=))
import qualified Data.ByteString.Lazy
import GHC.Generics (Generic)
import "base" Data.Maybe (fromMaybe)

-- | A complex record using many qualified imports
data ComplexData = ComplexData
  { text :: T.Text
  , mapping :: Map.Map T.Text Int
  , httpClient :: HTTP.Manager
  , vector :: Data.Vector.Vector Int
  , binary :: Data.ByteString.Lazy.ByteString
  }
  deriving (Generic, Show)
  deriving anyclass (ToJSON, FromJSON)

-- | Create an empty complex data structure
emptyComplex :: HTTP.Manager -> ComplexData
emptyComplex mgr = ComplexData
  { text = T.empty
  , mapping = Map.empty
  , httpClient = mgr
  , vector = Data.Vector.empty
  , binary = Data.ByteString.Lazy.empty
  }