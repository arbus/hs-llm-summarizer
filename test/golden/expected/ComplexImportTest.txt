-- Module: test/golden/input/ComplexImportTest.hs

 | A complex record using many qualified imports

data ComplexData = ComplexData{text :: T.Text,
                               mapping :: Map.Map T.Text Int, httpClient :: HTTP.Manager,
                               vector :: Data.Vector.Vector Int,
                               binary :: Data.ByteString.Lazy.ByteString}
                     deriving (Generic, Show)
                     deriving anyclass (ToJSON, FromJSON)

 | Create an empty complex data structure

emptyComplex :: HTTP.Manager -> ComplexData

